aspnetcore/blazor/hybrid/handle-errors.md

```diff
@@ -0,0 +1,82 @@
+uid: blazor/hybrid/handle-errors
+zone_pivot_groups: blazor-hybrid-operating-systems
+---
+# Handle errors in ASP.NET Core Blazor Hybrid
+
+This article explains how to use [browser developer tools](https://developer.mozilla.org/docs/Glossary/Developer_Tools) with Blazor Hybrid apps.
+
+[!INCLUDE[](~/blazor/includes/blazor-hybrid-preview-notice.md)]
+
+## Developer tools with .NET MAUI Blazor
+
+:::zone pivot="android"
+
+To use browser developer tools with an Android app:
+
+1. Start the Android emulator and navigate to an app page that uses a `BlazorWebView`.
+1. Open Google Chrome or Microsoft Edge.
+1. Navigate to `chrome://inspect/#devices` (Google Chrome) or `edge://inspect/#devices` (Microsoft Edge).
+1. Select the **`inspect`** link button to open developer tools. The following example shows the **DevTools** page in Microsoft Edge:
+
+   ![Microsoft Edge Devices showing the BlazorWebView's "inspect" link button to open developer tools.](~/blazor/hybrid/handle-errors/_static/android.png)
+
+1. Developer tools provide a variety of features for working with apps, including which assets the page requested, how long assets took to load, and the content of loaded assets. The following example shows the **Console** tab to see the console messages, which includes any exception messages generated by the framework or developer code:
+
+   ![Microsoft Edge DevTools window for a Blazor Hybrid app running on an emulated Pixel 5](~/blazor/hybrid/handle-errors/_static/edge1.png)
+
+:::zone-end
+
+:::zone pivot="ios"
+
+To use Safari developer tools with an iOS app:
+
+1. Open desktop Safari.
+1. Select the **Preferences** > **Advanced** > **Show Develop** menu item.
+1. Run the .NET MAUI Blazor app in the iOS simulator.
+1. Return to desktop Safari. Select **Develop** > **Simulator** > **0.0.0.0**. If multiple entries for **0.0.0.0** are present, select the entry that highlights the `BlazorWebView`. The `BlazorWebView` is highlighted in blue in the iOS simulator when the correct **0.0.0.0** entry is selected.
+
+   ![Safari Develop Simulator open showing two entries for "0.0.0.0" with the second entry selected because it highlights the BlazorWebView in the Visual Studio emulator UI.](~/blazor/hybrid/handle-errors/_static/ios.png)
+
+1. The **Web Inspector** window appears for the `BlazorWebView`.
+1. Developer tools provide a variety of features for working with apps, including which assets the page requested, how long assets took to load, and the content of loaded assets. The following example shows the **Console** tab, which includes any exception messages generated by the framework or developer code:
+
+   ![Safari Web Inspector and Simulator windows for a Blazor Hybrid app running on an emulated iPad mini](~/blazor/hybrid/handle-errors/_static/safari1.png)
+
+:::zone-end
+
+:::zone pivot="macos"
+
+To use Safari developer tools with a macOS app:
+
+1. Open desktop Safari.
+1. Select the **Preferences** > **Advanced** > **Show Develop** menu item.
```

Comment:
Was just trying this out and I was mistaken there is a bit more nuance than just the iOS steps. Looking into this and will update...

Reply:
Looks like this may be a regression on Mac/Safari: https://apple.stackexchange.com/questions/416277/how-to-inspect-safari-web-inspector-aka-devtools-on-devtools

Looking for workarounds...

Reply:
Okay this looks like a genuine bug. Can we just remove the macos section entirely for now?

Bug: https://github.com/dotnet/maui/issues/5561